File: LeetCode/Python Solutions/Kth Distinct String in an Array/kth.py
from collections import Counter

class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        # Count occurrences of each string
        count = Counter(arr)
        
        # Find the kth distinct string
        distinct_count = 0
        for s in arr:
            if count[s] == 1:
                distinct_count += 1
                if distinct_count == k:
                    return s
        
        # If there are fewer than k distinct strings, return ""
        return ""

File: LeetCode/Java Solutions/Kth Distinct String in an Array/kth.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String kthDistinct(String[] arr, int k) {
        // Count occurrences of each string
        Map<String, Integer> count = new HashMap<>();
        for (String s : arr) {
            count.put(s, count.getOrDefault(s, 0) + 1);
        }
        
        // Find the kth distinct string
        int distinctCount = 0;
        for (String s : arr) {
            if (count.get(s) == 1) {
                distinctCount++;
                if (distinctCount == k) {
                    return s;
                }
            }
        }
        
        // If there are fewer than k distinct strings, return ""
        return "";
    }
}